### PR TITLE
FEATURE: QA-17152 Locale setting Cypress test for insightView and dashboardComponent

### PR DESCRIPTION
**Pull request by NghiLamGD**
JIRA: QA-17152

---

Supported PR commands:

| Command                                                 | Description                                                |
| ------------------------------------------------------- | ---------------------------------------------------------- |
| `ok to test`                                            | Re-run standard checks                                     |
| `extended check sonar`                                  | SonarQube tests                                            |
| `extended check plugins`                                | Dashboard plugins tests                                    |
| `extended test - backstop`                              | BackstopJS tests                                           |
| **E2E Cypress tests commands - TIGER**                  |                                                            |
| `extended test - tiger-cypress - isolated <testName>`   | Run isolated tests running against recorded Tiger backend. |
| `extended test - tiger-cypress - record <testName>`     | Create a new recording for isolated Tiger tests.           |
| `extended test - tiger-cypress - integrated <testName>` | Run integrated tests against live backend                  |
| **E2E Cypress tests commands - BEAR**                   |                                                            |
| `extended test - cypress - isolated <testName>`         | Run isolated tests running against recorded Bear backend.  |
| `extended test - cypress - record <testName>`           | Create a new recording for isolated Bear tests.            |

`<testName>` in cypress commands is used to filter specfiles. Example, to run record with BEAR backend

-   Against `dashboard.spec.ts` and `drilling.spec.ts`, execute command `extended test - cypress - record dashboard,drilling`
-   Against all specfiles, execute command `extended test - cypress - record` or `extended test - cypress - record *`

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `extended test - backstop` passes
-   [ ] `extended test - tiger-cypress - record` to record new mapping files (Tiger BE)
-   [ ] `extended test - cypress - record` to record new mapping files (Bear BE)
-   [ ] `extended test - tiger-cypress - isolated` passes
-   [ ] `extended test - cypress - isolated` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/11612/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/13670/ : FAILURE in 6m 25s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/11613/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/13671/ : FAILURE in 6m 39s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/11615/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/13672/ : SUCCESS in 6m 40s

**Comment by hungnguyenauto** [comment]
LGTM

File: libs/sdk-ui-tests-e2e/cypress/tools/apiRequest/LocalizationApi.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
+// (C) 2022-2023 GoodData Corporation
+import { getHost, getTigerAuthToken, getProjectId } from "../../support/constants";
+
+class LocalizationApi {
+    localeSettingViaAPI = (value: string, level: string) => {
+        let url: string;
+        if (level === "userSetting") {
+            cy.request({
+                method: "GET",
+                url: `${getHost()}/api/v1/profile`,
```
Comment (kandl) [inline libs/sdk-ui-tests-e2e/cypress/tools/apiRequest/LocalizationApi.ts:10]: Is it possible to use `api-client-tiger` for these requests instead?

Comment (NghiLamGD) [reply libs/sdk-ui-tests-e2e/cypress/tools/apiRequest/LocalizationApi.ts:10]: I haven't tried using `api-client-tiger` yet. Do you have any existing test which use it? please send it to me

Comment (kandl) [reply libs/sdk-ui-tests-e2e/cypress/tools/apiRequest/LocalizationApi.ts:10]: I think it should be possible to process all these requests with usage of `api-client-tiger`, not sure if it's used anywhere in the tests now. Or in the best case, can we use normal `sdk-backend-tiger` instance and its methods? There is setLocale method if I'm correct.

Comment (NghiLamGD) [reply libs/sdk-ui-tests-e2e/cypress/tools/apiRequest/LocalizationApi.ts:10]: Screen Shot 2023-02-28 at 17 09 05

I tried 
`import sdk from "@gooddata/api-client-tiger";
sdk.entities....`
but I can't find setLocale or updateLocale or anything related to locale (I can find theme/colorpalette,... but with locale I can't)

Comment (kandl) [reply libs/sdk-ui-tests-e2e/cypress/tools/apiRequest/LocalizationApi.ts:10]: setLocale method is in `sdk-backend-tiger`

Comment (NghiLamGD) [reply libs/sdk-ui-tests-e2e/cypress/tools/apiRequest/LocalizationApi.ts:10]: IMO, i don't think `sdk-backend-tiger` should be used in this test. 
This test is based on real user actions, (refer to https://www.gooddata.com/developers/cloud-native/doc/cloud/customize-appearance/change-display-language/), verify locale is applied if sending API requests directly as the document mentioned

**Pull request closed without merging**